shell/components/form/LabeledSelect.vue

```diff
@@ -159,6 +148,13 @@ export default {
         return;
       }
 
+      // Force to update the option label if prop has been changed
+      const isOutdated = !this.options.find(({ label }) => option.label === label);
```

Comment:
It's possible to work with options that do not have `label` and `value` properties by setting the `optionLabel` and `optionValue` props, so I think this needs to use eg 
`!this.options.find((opt) => option[this.optionLabel]===opt[this.optionLabel]);`